packages/arb-token-bridge-ui/src/hooks/useTransactionHistory.ts

```diff
@@ -0,0 +1,676 @@
+} from '../state/app/utils'
+import {
+  EthWithdrawal,
+  isTokenWithdrawal,
+  mapETHWithdrawalToL2ToL1EventResult,
+  mapTokenWithdrawalFromEventLogsToL2ToL1EventResult,
+  mapWithdrawalToL2ToL1EventResult
+} from '../util/withdrawals/helpers'
+import { FetchWithdrawalsFromSubgraphResult } from '../util/withdrawals/fetchWithdrawalsFromSubgraph'
+import { updateAdditionalDepositData } from '../util/deposits/helpers'
+import { useCctpFetching } from '../state/cctpState'
+import {
+  getDepositsWithoutStatusesFromCache,
+  getProvider,
+  getUpdatedCctpTransfer,
+  getUpdatedEthDeposit,
+  getUpdatedTokenDeposit,
+  getUpdatedWithdrawal,
+  isSameTransaction,
+  isTxPending
+} from '../components/TransactionHistory/helpers'
+import { useIsTestnetMode } from './useIsTestnetMode'
+
+export type TransactionHistoryParams = {
+  data: {
+    transactions: MergedTransaction[]
+    numberOfDays: number
+  }
+  loading: boolean
+  completed: boolean
+  error: unknown
+  failedChainPairs: ChainPair[]
+  pause: () => void
+  resume: () => void
+  addPendingTransaction: (tx: MergedTransaction) => void
+  updatePendingTransaction: (tx: MergedTransaction) => void
+}
+
+export type ChainPair = { parentChain: ChainId; chain: ChainId }
+
+export type Deposit = Transaction
+
+export type Withdrawal =
+  | FetchWithdrawalsFromSubgraphResult
+  | WithdrawalInitiated
+  | EthWithdrawal
+
+type DepositOrWithdrawal = Deposit | Withdrawal
+type Transfer = DepositOrWithdrawal | MergedTransaction
+
+function getStandardizedTimestampByTx(tx: Transfer) {
+  if (isCctpTransfer(tx)) {
+    return (tx.createdAt ?? 0) / 1_000
+  }
+
+  if (isDeposit(tx)) {
+    return tx.timestampCreated ?? 0
+  }
+
+  if (isWithdrawalFromSubgraph(tx)) {
+    return getStandardizedTimestamp(tx.l2BlockTimestamp)
+  }
+
+  return getStandardizedTimestamp(tx.timestamp ?? '0')
+}
+
+function sortByTimestampDescending(a: Transfer, b: Transfer) {
+  return getStandardizedTimestampByTx(a) > getStandardizedTimestampByTx(b)
+    ? -1
+    : 1
+}
+
+const multiChainFetchList: ChainPair[] = [
+  {
+    parentChain: ChainId.Ethereum,
+    chain: ChainId.ArbitrumOne
+  },
+  {
+    parentChain: ChainId.Ethereum,
+    chain: ChainId.ArbitrumNova
+  },
+  // Testnet
+  {
+    parentChain: ChainId.Goerli,
+    chain: ChainId.ArbitrumGoerli
+  },
+  {
+    parentChain: ChainId.Sepolia,
+    chain: ChainId.ArbitrumSepolia
+  },
+  // Orbit
+  {
+    parentChain: ChainId.ArbitrumGoerli,
+    chain: ChainId.XaiTestnet
+  },
+  {
+    parentChain: ChainId.ArbitrumSepolia,
+    chain: ChainId.StylusTestnet
+  },
+  ...getCustomChainsFromLocalStorage().map(chain => {
+    return {
+      parentChain: chain.partnerChainID,
+      chain: chain.chainID
+    }
+  })
+]
+
+function isWithdrawalFromSubgraph(
+  tx: Withdrawal
+): tx is FetchWithdrawalsFromSubgraphResult {
+  return tx.source === 'subgraph'
+}
+
+function isDeposit(tx: DepositOrWithdrawal): tx is Deposit {
+  return tx.direction === 'deposit'
+}
```

Comment:
I will be removing old utils that were used by the old tx history, will be done in new PRs as these clean ups are non blocking

Reply:
`isDeposit` in this file is comparing fetched txs, not merged txs. It will be part of the refactoring to completely unify all the types of txs so we have a single source for them